### ISSUE TITLE
Build fails due to glxext.h on Linux

### ISSUE THREAD

**Issue by whg** [cinder/Cinder#2064]
Cinder fails to build due to some conflicting types and I get the following:
```
In file included from /home/whg/Code/Cinder/include/glload/glx_all.h:6,
                 from /home/whg/Code/Cinder/src/glload/glx_load.c:4:
/home/whg/Code/Cinder/include/glload/_int_glx_type.h:96:3: error: conflicting types for ‘GLXHyperpipeNetworkSGIX’
 } GLXHyperpipeNetworkSGIX;
   ^~~~~~~~~~~~~~~~~~~~~~~
```
The culprits are:
* `GLXHyperpipeNetworkSGIX`
* `GLXHyperpipeConfigSGIX`
* `GLXPipeRect`
* `GLXPipeRectLimits`
* `PFNGLXGETCURRENTDISPLAYEXTPROC`
* `PFNGLXGETCURRENTREADDRAWABLESGIPROC`

I'm on Linux and using Mesa (version 18.2.6).

I've made a temporary fix by putting some include guards in the glx source:
[https://github.com/whg/Cinder/commit/acdf31312e88332e2f22e2a7609597d31d6a2f8f](https://github.com/whg/Cinder/commit/acdf31312e88332e2f22e2a7609597d31d6a2f8f)

But it's probably not the way to go! I checked out the recent commits on Mesa but couldn't find anything suspect, so I'm slightly confused about why it's happening.

**Comment by gaborpapp** [comment]
Also reported in issue #2062 
Are you also on Arch?

**Comment by whg** [comment]
No, I'm on Manjaro, but will be getting the same packages, so no doubt it's the same problem.

After a bit of testing I found that it was the upgrade from Mesa 18.2.4 to 18.2.6 that broke the build. After checking the diffs I found the offending line. In a [recent commit](https://gitlab.freedesktop.org/mesa/mesa/commit/f7d42ee7d319256608ad60778f6787c140badada#31ac431cdd8c73c76ffa76809620c3e2c0ada784_2_1), a define `/usr/include/GL/glxext.h` changed from: 

`#define __glxext_h_` 
to 
`#define __glx_glxext_h_`

This changes breaks `glload` because it tests agains `__glxext_h_`.

@gaborpapp it's interesting to hear to you're using a recent Mesa and it compiles for you. What's defined in your `/usr/include/GL/glxext.h`?

I've made a cleaner fix to resolve the build here:  [https://github.com/whg/Cinder/commit/4a7756aeab15a2e89f38ed313fb87cc37dab087e](https://github.com/whg/Cinder/commit/4a7756aeab15a2e89f38ed313fb87cc37dab087e). Let me know if you think it's worth making a pull request with it.

**Comment by PetrosKataras** [comment]
A PR with the fix would be great - Seems to be the way to go for providing back/forward compatibility.

**Comment by nanu-c** [comment]
@whg  your merge should be added also to `include/glload/_int_glx_type.hpp`

**Comment by whg** [comment]
Thanks!
